test/cljx/schema/core_test.cljx

```diff
@@ -229,6 +239,21 @@
     (is (= '(either {:a Int} Int) (s/explain (s/either {:a s/Int} s/Int))))
     (is (s/explain schema))))
 
+(deftest constrained-test
+  (let [schema (s/constrained
+                {s/Keyword s/Keyword}
+                (fn [m] (every? (fn [[k v]] (= k v)) m))
+                'equal-keys?)]
+    (valid! schema {})
+    (valid! schema {:foo :foo :bar :bar})
+    (invalid! schema {"foo" "foo"})
+    (invalid! schema {:foo :bar} "(not (equal-keys? {:foo :bar}))")
+    (invalid! schema {:foo 1} "(not (equal-keys? {:foo 1}))")
```

Comment:
i'm not totally sure. possibly something like spec/precondition instead of the predicates in the conditional? i guess that would be a breaking change.

not sure if it's too dirty, but if there is an odd number of args to the conditional, the last one is the error message?